samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -106,4 +112,25 @@ final ExecutionPlan getExecutionPlan(StreamApplication app) throws Exception {
   final StreamManager getStreamManager() {
     return streamManager;
   }
+
+  /**
+   * Write the execution plan JSON to a file
+   * @param planJson JSON representation of the plan
+   */
+  final void writePlanJsonFile(String planJson) {
```

Comment:
I'd prefer logging the plan and allowing user to copy paste it into the viewer instead of creating files on disk ourselves. This environment variable might not exist in standalone for example. If users want a separate logger/file for this they can do that in their log4j.xml.